frontend/src/metabase/lib/hooks.js

```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect, useMemo, useState } from "react";
+
+export function useAsyncFunction(fn) {
+  const [isPending, setIsPending] = useState(false);
+  const isUnmountedRef = useRef(false);
+
+  useEffect(() => {
+    () => {
+      isUnmountedRef.current = true;
+    };
+  }, []);
+
+  const safeFn = useMemo(() => {
+    return (...args) =>
+      new Promise((resolve, reject) => {
+        setIsPending(true);
+        return fn(...args)
+          .then(res => {
+            if (!isUnmountedRef.current) {
+              resolve(res);
+            }
+            setIsPending(false);
```

Comment:
This `setIsPending(false)` should be inside `if` otherwise it might be executed on an unmounted component which will trigger an error

Reply:
oh derp, didn't think that one through. fixed